src/lib/utils.ts

```diff
@@ -0,0 +1,124 @@
+import type { JSONSchema, SchemaObject } from 'oas/dist/rmoas.types';
+
+import { get as lodashGet } from 'lodash'; // eslint-disable-line no-restricted-imports
+
+/**
+ * Determine if a schema `type` is, or contains, a specific discriminator.
+ *
+ */
+export function hasSchemaType(
+  schema: SchemaObject,
+  discriminator: 'array' | 'object' | 'string' | 'number' | 'boolean' | 'integer' | 'null'
+) {
+  if (Array.isArray(schema.type)) {
+    return schema.type.includes(discriminator);
+  }
+
+  return schema.type === discriminator;
+}
+
+/**
+ * With a supplied JSON Schema object, spider through it for any schemas that may contain specific
+ * kind of `format` that also happen to be within the current `requestBody` payload that we're
+ * creating a HAR representation for.
+ *
+ */
+export function getTypedFormatsInSchema(
```

Comment:
With the test I wrote in `requestBody.test.ts` this thing ends up returning an array that looks like:

```js
['root', 'obj.prop', 'arr.0', 'arrOfObjects.0.arrProp', 'arrOfObjects.1.arrProp']
```

And because that's dot-notation we can easily use Lodash `get` and `set` to manipulate that data into JSON objects.